DeAceroDLabs/marketplace-frontend
# Feature/dmp 601 Dockerfile

## What is done

[[DMP-601] Dockerfile](https://deacero.atlassian.net/browse/DMP-601)

## How is done

- Created a `Dockerfile` and `dockerignore` file.
- Added host variable on setProxy file.
- At the moment we do not have the back on a container, that is why we need to change manually host variable on  setProxy file.
- when using docker container const host = "host.docker.internal" else const host = "localhost" on setProxy file.
- Added localhost/8200 and localhost/8300 in Auth0 (Applications/employees-store) so we can access with Auth0.
![image](https://user-images.githubusercontent.com/81311646/190023216-2359b0dd-056e-4dbc-aaf2-7466c12637cf.png)

- To delete this comment, we need to create the dockerfile on back:
![image](https://user-images.githubusercontent.com/81311646/190023611-71b7081c-a8a6-4776-9fb1-4c8df5c3fd59.png)



## How to test

1. Run  `npm run docker:build`
2. Then run  `npm run docker:run`
3. Plase wait so that the container runs the react app correctly. You can see the process with `npm run docker:logs`
4. Open http://localhost:8300/ to see the container.
5. You should see the Home Screen or login. And everything works correctly.

### To open project locally on development (the way we always do it) parallely with container:

1. On setProxy file, set the host variable to localhost,  const host = "localhost"
2. Run `docker run -p 8200:8300 --rm -d -it --name front-market mkt-place-front:dev` 
3. npm start
4. http://localhost:8300/ To see local project.
5. http://localhost:8200/ To see container project. (Note: signup only works now with 8300 because Auth0)
6. Everythings works correctly outside container (expect signup).

## To stop container and remove it
Run  `npm run docker:stop`
